src/object/icecrusher.cpp

```diff
@@ -226,6 +333,40 @@ IceCrusher::update(float dt_sec)
         else
           m_col.m_movement = Vector (0, RECOVER_SPEED_NORMAL);
       }
+      break;
+	case RECOVERING_RIGHT:
+      if (m_col.m_bbox.get_left() <= start_position.x+1) {
+        set_pos(start_position);
+        m_col.m_movement = Vector (0, 0);
+        if (ic_size == LARGE)
+          cooldown_timer = PAUSE_TIME_LARGE;
+        else
+          cooldown_timer = PAUSE_TIME_NORMAL;
+        set_state(IDLE);
+      }
+      else {
+        if (ic_size == LARGE)
+          m_col.m_movement = Vector (RECOVER_SPEED_LARGE, 0);
+        else
+          m_col.m_movement = Vector (RECOVER_SPEED_LARGE, 0);
+      }
+      break;
+	case RECOVERING_LEFT:
+      if (m_col.m_bbox.get_left() >= start_position.x-1) {
+        set_pos(start_position);
+        m_col.m_movement = Vector (0, 0);
+        if (ic_size == LARGE)
+          cooldown_timer = PAUSE_TIME_LARGE;
+        else
+          cooldown_timer = PAUSE_TIME_NORMAL;
+        set_state(IDLE);
+      }
+      else {
+        if (ic_size == LARGE)
```

Comment:
The code in both if and else is the same.

Reply:
Fixed.